app/script/links/LinkPreviewHelpers.js

```diff
@@ -41,12 +41,12 @@ z.links.LinkPreviewHelpers = {
    * @returns {Array<string, number>} First link and its offset
    */
   get_first_link_with_offset(text) {
-    const links = twttr.txt.extractUrls(text);
+    const links = linkify.find(text, 'url');
     const first_link = links[0];
 
     if (first_link != null) {
-      const link_offset = text.indexOf(first_link);
-      return [first_link, link_offset];
+      const link_offset = text.indexOf(first_link.value);
+      return [first_link.value, link_offset];
```

Comment:
Can we turn this into a more explicit object instead of hoping to remember in which order these arguments are in the array?